src/Orchard.Web/Modules/Orchard.Users/Services/UserService.cs

```diff
@@ -85,7 +86,7 @@ ISiteService siteService
         public string CreateNonce(IUser user, TimeSpan delay) {
             var challengeToken = new XElement("n", new XAttribute("un", user.UserName), new XAttribute("utc", _clock.UtcNow.ToUniversalTime().Add(delay).ToString(CultureInfo.InvariantCulture))).ToString();
             var data = Encoding.UTF8.GetBytes(challengeToken);
-            return Convert.ToBase64String(_encryptionService.Encode(data));
+            return HttpServerUtility.UrlTokenEncode(_encryptionService.Encode(data));

```

Comment:
If I make the suggested code compile-able, I would end up with something like this.

``return 
HttpServerUtility.UrlTokenEncode(Convert.FromBase64String(Convert.ToBase64String(_encryptionService.Encode(data))));``
and
``var data = _encryptionService.Decode(Convert.FromBase64String(Convert.ToBase64String(HttpServerUtility.UrlTokenDecode(nonce))));``

That's repetitive conversion string to bytes and vice versa. Please bear with me, I am really missing something here.